addons/stock/wizard/stock_quant_relocate.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-is_not_full_package
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+class RelocateStockQuant(models.TransientModel):
+    _name = 'stock.quant.relocate'
+    _description = 'Stock Quantity Relocation'
+
+    quant_ids = fields.Many2many('stock.quant')
+    company_id = fields.Many2one(related="quant_ids.company_id")
+    location_id = fields.Many2one('stock.location', domain="[('usage', '=', 'internal'), ('company_id', '=', company_id)]")
+    allowed_package_ids = fields.Many2many('stock.quant.package', compute="_compute_allowed_package_ids")
+    package_id = fields.Many2one('stock.quant.package', domain="[('id', 'in', allowed_package_ids)]")
+    message = fields.Text('Reason for relocating stock')
+    is_not_full_package = fields.Boolean(compute='_compute_is_not_full_package')
+    is_not_single_location = fields.Boolean(compute='_compute_is_not_single_location')
+
+    def process(self):
+        if not self.location_id and not self.package_id:
+            return
+        self.quant_ids.action_set_inventory_quantity_to_zero()
+        # if quants have reserved quantities and are in a package, we have to unreserve the related stock.move.line
+        # because the package is removed before creating the move, which causes issues with reserved quantities in _free_reservation()
+        reserved_quants = self.quant_ids.filtered(lambda q: q.reserved_quantity > 0)
+        reserved_quants._unreserve_quants()
+
```

Comment:
You could remember the moves you unreserve and reserve them again at the end (NTH)